specificatie/openapi.yaml

```diff
@@ -207,17 +207,36 @@ paths:
           $ref: 'https://raw.githubusercontent.com/VNG-Realisatie/Haal-Centraal-common/v1.2.0/api-specificatie/common.yaml#/components/responses/default'
 
   /adresseerbareobjecten:
+    #7, 331, 332, 334, 373
     get:
       security:
         - apiKeyBAG: []
       operationId: raadpleegAdresseerbareobjecten
       tags:
         - Adresseerbaar object
       summary: "vindt verblijfsobjecten, ligplaatsen, standplaatsen"
-      description: "Zoek actuele adresseerbare objecten (verblijfsobjecten, standplaatsen of ligplaatsen) met een nummeraanduidingidentificatie of pandidentificatie. Gebruik de fields-parameter als je alleen specifieke velden in het antwoord wil zien, zie [functionele specificaties fields-parameter](https://github.com/VNG-Realisatie/Haal-Centraal-common/blob/v1.2.0/features/fields.feature). Gebruik de expand-parameter als je het antwoord wil uitbreiden met (delen van) de gerelateerde resources adres en pand, zie [functionele specificaties expand-parameter](https://github.com/VNG-Realisatie/Haal-Centraal-common/blob/v1.2.0/features/expand.feature)."
+      description: "Zoek actuele adresseerbare objecten (verblijfsobjecten, standplaatsen of ligplaatsen):<br>
+        1. met een nummeraanduidingidentificatie of<br>
+        2. een pandidentificatie of<br> 
+        3. met een status geconstateerd in combinatie met woonplaats identificatie aangevuld met type en één of meer gebruiksdoelen of<br>
+        4. met een geometrische contour (punt, rechthoek of cirkel) in combinatie met status geconstateerd, oppervlakte, gebruiksdoel en/of type.<br>
+        Gebruik de fields-parameter als je alleen specifieke velden in het antwoord wil zien, zie 
+        [functionele specificaties fields-parameter](https://github.com/VNG-Realisatie/Haal-Centraal-common/blob/v1.2.0/features/fields.feature). 
+        Gebruik de expand-parameter als je het antwoord wil uitbreiden met (delen van) de gerelateerde resources adres en pand, zie 
+        [functionele specificaties expand-parameter](https://github.com/VNG-Realisatie/Haal-Centraal-common/blob/v1.2.0/features/expand.feature)."
       parameters:
         - $ref: '#/components/parameters/nummeraanduidingIdentificatie-query'
         - $ref: '#/components/parameters/pandIdentificatie-query'
+        - $ref: '#/components/parameters/geconstateerd'
+        - $ref: '#/components/parameters/woonplaatsIdentificatie-query'
+        - $ref: '#/components/parameters/oppervlakte'
+# oppervlakte toevoegen?
+        - $ref: '#/components/parameters/type'
+        - $ref: '#/components/parameters/gebruiksdoelen'
+        - $ref: '#/components/parameters/locatie-query'
+# zie opmerking over locatie-query bij /panden
+        - $ref: '#/components/parameters/bbox'
+        - $ref: '#/components/parameters/cirkel'
```

Comment:
Een check, de cirkel parameter is een object. Dit betekent dat er json in de query string wordt gestopt, dus zoiets: /adresseerbareobjecten?punt={"coordinates":[34587136.729115635,-82836734.02848643],"type":"Point"} (gegenereerd door postman). Dit geldt ook voor bbox. Willen we dit?

Reply:
@MelvLee wat jij hier opschrijft ziet er ineens heel anders uit. Best wel complex met al die haakjes en accolades en dubbele quotes. Niet echt de querystring die wij nog als uitbreiding van een eenvoudige REST API voor ogen hadden, toch @allen?

Reply:
@Melvin, wat stel jij je voor bij een eenvoudige querystring parameter?

Reply:
Volgens mij is het zelfs zo dat als je geen gebruik wil maken van de default radius de query string als volgt wordt:

`/adresseerbareobjecten?cirkel={"coordinates":[34587136.729115635,-82836734.02848643],"type":"Point","radius":200}`

We zouden de parameter 'cirkel' als een array van 3 waarden kunnen definiëren (2 voor de coördinaten en 1 optionele voor de radius), dus als volgt:

```
  cirkel:
    description: "Cirkel waarbinnen het object dat je zoekt valt. Objecten die de rand van de cirkel raken worden in het zoekresultaat meegenomen."
     name: cirkel
     in: query
     required: false
     schema:
         type: array
         minItems: 2
         maxItems: 3
         items:
            type: number 
```

al is dit wat minder zelfbeschrijvend. De query komt er dan als volgt uit te zien:

`/adresseerbareobjecten?cirkel=[34587136.729115635,-82836734.02848643,200]`

Reply:
Voor simpele geometrie zoeken zou ik ervoor kiezen om alleen de bounding box query parameter te introduceren. Met een bounding box kan je een punt aangeven als de linkerbovenpunt en de rechteronderpunt dezelfde waarde hebben. Een cirkel kan je in een bounding box omvatten door de zijden even lang te maken (ro-x - lb-x = ro-y - lb-y). Je zou mogelijk te veel resultaat krijgen (in de hoeken van de bbox) maar dit is gerechtvaardigd (denk ik) omdat het om simpele zoeken gaat

Reply:
Hiermee kunnen we ook de locatie query parameter deprecaten

Reply:
Maak dit het dan niet complexer ? Een developer moeten we nu uit gaan leggen dat als hij een punt met een radius op wil geven dat hij daar dan de bounding Box voor moet "misbruiken". Het scheelt wel een property, maar we maken er wel meer een puzzel van.

Reply:
Waarom definiëren we geen parameter 'area' met als type een polygon. De applicatie waarin de medewerker van de pandbrigade aangeeft in welk gebied hij wil zoeken kan voorzien in de vormen waarin wij nu in de API voorzien, dus cirkel, boundingbox en punt, en zelfs meer (zoals een willekeurige geschetste vorm) als men daar behoefte aan heeft. Omrekening van die vorm naar polygon dient dan in de betreffende applicatie te gebeuren.
Resultaat maximale flexibiliteit en een eenvoudige API.

Nadeel is dat de request url wel erg groot kan worden en dat je 'punt' op de e.o.a. wijze moet vertalen naar een polygon. Dat laatste lijkt me echter niet zo'n groot probleem.

Reply:
Dat laatste nadeel dat je meldt is precies de redenwaarom we beperkingen willen leggen om de wat er in de request-url tercht kan komen.

Reply:
> Nog een punt. Er is voor gekozen om de nieuwe parameters tussen de bestaande parameters toe te voegen. Als een consumer al een implementatie heeft gemaakt en hij genereert code op basis van deze yaml, dan leidt dit mogelijk tot compile errors. Dit omdat het signatuur van de methodes zijn veranderd. Dit is in principe wat er ook gebeurt als een parameter wordt verwijderd of als een verplichte parameter wordt toegevoegd. Afhankelijk van de programmeer stijl kan dit leiden tot veel aanpassingen en kan dus gezien worden als een breaking change. Willen we dit voorkomen, dan moeten de nieuwe parameters achter de bestaande parameters worden toegevoegd.

Begrijp ik dit goed?
Je wilt wel verbeteringen op bestaande functionaliteit (opruimen e.d.), maar geen nieuwe functionaliteit implementeren. Dat betekent dat je bestaande code opnieuw genereert EN NIET gaat ontwikkelen/programmeren. Jij zegt dus dat dit geen patch is (die je gewoon kunt toevoegen), maar dat er iets veranderd is. Je moet dus regressietesten doen, en je wordt gedwongen om je code aan te passen. Het feit dat je dit moet, in jouw ogen een breaking change. Begrijp ik dit goed?

Reply:
En ten tweede: als je deze nieuwe parameters achteraan toevoegt gaat het wel goed, en zijn maw geen codeeraanpassingen nodig?

Reply:
Ik zet de nieuwe parameters in de specificatie onder de reeds gedefinieerd parameters, dan ontstaat er geen breaking change.

Reply:
Punt en Cirkel zijn verwijderd. Bounding box en locatie blijven over. Voor bbox wordt altijd een intersects gebruikt (volgens OGC API Features). Intersects is beschreven als intersects (geom1, geom2) = ! geom1 disjount geom2, een disjount betekent dat geom1 en geom2 niet raken, kruisen, overlappen of bevatten. Dus het tegenovergestelde ervan (!) is dat als ze wel raken, kruisen, overlappen of bevatten, dat ze intersecten. Als er een punt wordt meegegeven voor de bbox (lo=rb) en deze ligt binnen de geometrie van bv. een pand, of op de rand, dan levert een intersects een true op omdat de geometrieën overlappen of raken en een query levert in dat geval het pand op.

Reply:
Dank voor het uitzoeken!